src/Processors/QueryPlan/Optimizations/optimizePrimaryKeyConditionAndLimit.cpp

```diff
@@ -18,7 +18,8 @@ void optimizePrimaryKeyConditionAndLimit(const Stack & stack)
     const auto & storage_prewhere_info = source_step_with_filter->getPrewhereInfo();
     if (storage_prewhere_info)
     {
-        source_step_with_filter->addFilter(storage_prewhere_info->prewhere_actions.clone(), storage_prewhere_info->prewhere_column_name);
+        if (storage_prewhere_info->prewhere_actions)
+            source_step_with_filter->addFilter(storage_prewhere_info->prewhere_actions->clone(), storage_prewhere_info->prewhere_column_name);
         if (storage_prewhere_info->row_level_filter)
             source_step_with_filter->addFilter(storage_prewhere_info->row_level_filter->clone(), storage_prewhere_info->row_level_column_name);
     }
```

Comment:
To me this code looks rather warring (don't mean actual change).
It seems that some conditions like Limit applied before row policy. Is it true? If yes, is it risky from security standpoint?